tests/frontend/test_program.py

```diff
@@ -477,18 +442,22 @@ def test_user_defined_decomposition_false(self):
 
         In this case, the Gaussian operation should remain after compilation.
         """
-        prog = sf.Program(3)
-        cov = np.array([[0, 1], [1, 0]])
-
+        prog = sf.Program(2)
+        cov = random_covariance(2, pure=False)
+        r = np.random.randn(4)
```

Comment:
A lot of the test modules have
```
# make test deterministic
np.random.random(42)
```
near the beginning of the module. I'm guessing the intent is to set the random seed to 42, in which case it should be `np.random.seed(42)`

Reply:
Oops, yes, that should definitely be `np.random.seed(42)`. Setting the random seed (or at least, we thought we were) was a stop-gap to force the tests to be deterministic, with the end goal of removing most calls to `np.random.random` for generating test data.

Reply:
I'll fix them.

Reply:
Done.

Reply:
@smite Remember to fix up here as well :+1:. In this case, just revert back to previous behaviour (adding `r` as well), because it doesn't seem that this test really cares about the value of arguments.